### DISCUSSION TITLE
Finding lab data for MIMIC-IV-ED patients?

### DISCUSSION THREAD

**Discussion by tchang1997** [MIT-LCP/mimic-code#1558] [MIMIC-IV-ED]
I am trying to work with lab data for patients in MIMIC-IV-ED. So far, I've been using the `labevents` table in MIMIC-IV. However, I've run into difficulties finding data for patients with a null `hadm_id`. I've tried to link such patients with the `labevents` table without much success — I describe my strategy in more depth here.

(1) For patients with a non-null hadm_id, obtaining their `labevents` info is straightforward since `labevents` stores the `hadm_id`. 
(2) I mostly have problems with patients with a null hadm_id. I tried to link `labevents` to ED stays as follows:
* For each `labevent`, retrieve all ED stays with a matching `subject_id`
* Find the ED stay such that the current lab's `charttime` falls between the `intime` and `outtime` attributes

In other words, I'm trying to find all lab tests done for a particular patient at any time, and filling in the `stay_id` for each lab event based on whether the `charttime` falls within any of that patient's ED stays. 

Based on this strategy, however, <1% of all patients with a null hadm_id are tested, which seems unreasonably low, even though we're looking at non-admitted patients. I get higher testing rates when linking to a test in `microbiologyevents` instead of `labevents,` but they still seem low. Is lab data for non-admitted patients in MIMIC-IV-ED available anywhere else in MIMIC? Or is there a more reliable strategy for linking labevents to ED stays?

**Comment by alistairewj** [comment]
I agree <1% feels too low. Probably just a slight mismatch in timing. I would add a fuzzy window of +- 8 hours to the intime/outtime for ED stays and see if you still have a very low number of labs. I would be interested to see the percentages if you post them here.

**Reply by SunnyFields2025** [reply]
I'm finding there is a strong relationship between acuity based on triage assessment.  For example, within 2 hours of arrival to the ED, I found lab data for 97% of those triaged as 1, 94% for those triaged as 2, and 76% of those triaged as 3. (My sample only includes patients with chief complaints of chest pain, shortness of breath, or neck/jaw/shoulder/arm pain.) Those triaged as 4 or lower have almost no data, but then no one is suspecting that person of even possibly having a heart attack, which is what I am looking for.  I also limited myself to 20 tests that would be relevant to evaluating a person with pain that might be associated with a heart attack.  And I only required them to have at least one test.  

I would be interested in hearing if the data have gotten better since 2023, or I just figured out an approach to using it? Regardless, 8 hours isn't realistic for machine learning models to detect time sensitive conditions.
